x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/search/WeightedTokensUtils.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml.search;
+
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.BoostQuery;
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.SearchExecutionContext;
+
+import java.io.IOException;
+import java.util.List;
+
+public final class WeightedTokensUtils {
+
+    public static Query queryBuilderWithAllTokens(List<WeightedToken> tokens, MappedFieldType ft, SearchExecutionContext context) {
```

Comment:
Could we update this method & `queryBuilderWithPrunedTokens` to return `QueryBuilder` instances instead? It has a couple advantages:

- We can use these methods directly when rewriting queries (where we mainly work with `QueryBuilder`s), making it easier to remove `WeightedTokensQueryBuilder` in the future.
- We can pass fewer args to this method because the `QueryBuilder` will be converted to a `Query` by the caller

Reply:
@Mikep86 I see the value of your suggestions and I agree they'd be good, but I'd like to keep that level of change outside the scope of this PR. What do you think?

Reply:
Sounds good to me :)